### PR TITLE
Update Checker

### PR DESCRIPTION
**Pull request by prophetl33t**

### CODE REVIEW

File: DataTool/ToolFlags.cs [hunk 1]
```diff
@@ -52,6 +52,10 @@ public class ToolFlags : ICLIFlags {
         [Alias("ex")]
         public bool Expert;
 
+        [CLIFlag(Default = false, Flag = "disable-upd-chk", Help = "Disables datatool update check (happens if CASC init fails)", Parser = new[] { "DataTool.Flag.Converter", "CLIFlagBoolean" })]
```
Comment (Js41637) [inline DataTool/ToolFlags.cs:55]: can we simplify this to `no-update-check` 😊

File: DataTool/Program.cs [hunk 2]
```diff
@@ -400,6 +407,25 @@ public static class Program {
...
+                            Logger.Warn("Core", $"Newer version of DataTool is available! Local version: {ver_local} Latest: {ver_remote}\nDownload latest build at {LastestBuildDownloadUrl}");
+                        }
+                    }
+                }
+            }
+            catch {
+                Logger.Warn("Core", $"Unable to get information about the latest version of DataTool.");
+            }
+        }
+
```
Comment (Js41637) [inline DataTool/Program.cs:428]: Not bad however could we change it to something like this
```cs
private static void CheckForUpdate() {
    try {
        Version localVersion = new Version(Assembly.GetExecutingAssembly().GetCustomAttribute<AssemblyFileVersionAttribute>()?.Version ?? string.Empty);
        if (Debugger.IsAttached || localVersion.Revision == 0) {
            return;
        }

        using (WebClient web = new WebClient()) {
            dynamic data = JObject.Parse(web.DownloadString(LastestBuildInfoUrl));
            Version remoteVersion = new Version(data.build.version.ToString());
            var buildSuccess = data.build.status == "success";

            if (remoteVersion > localVersion && buildSuccess) {
                Logger.Warn("Core", $"Newer version of DataTool is available! Local version: {localVersion} Latest: {remoteVersion}\nDownload latest build at {LastestBuildDownloadUrl}");
            }
        }
    } catch (Exception ex) {
        Logger.Debug("Core", $"Update check failed. {ex.Message}");
    }
}
```

**Review by Js41637 — CHANGES_REQUESTED** [review]

**Comment by prophetl33t** [comment]
So....... Is it okay?

**Comment by yretenai** [comment]
So I have a problem with update checking without explicitly allowing it, but this is also one of the biggest issues the tool faces.

Donno, I'm uncomfortable with it.

**Comment by prophetl33t** [comment]
We can set DisableUpdateCheck to true by default, but notify at startup that the user can enable update checking.

**Comment by prophetl33t** [comment]
Or ask the user for permission to check the update if CASC fails.

**Comment by Js41637** [comment]
I don't think auto checking is a big deal, its just a single request when the tools have already failed. As long as there's a flag to disable it then i think it's ok.

I'd like to change where it checks for an update though, i've changed how the tools error when there is an unsupported version, we should move the update check there instead of here as this would check on just any random error. Having it only check when we know the tools don't support a game version would be better.

**Pull request closed without merging**